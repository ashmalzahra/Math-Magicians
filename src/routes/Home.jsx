const Home = () => (
  <div>
    <h1>Math Magicians</h1>
    <h3>Welcome to our page</h3>
    <p />
  </div>
);

export default Home;
