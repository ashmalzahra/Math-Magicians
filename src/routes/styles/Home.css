.home-text {
  font-size: xx-large;
  padding-bottom: 1%;
  padding-left: 1.5%;
}

p {
  font-size: large;
  padding-bottom: 1%;
  padding-left: 1%;
  line-height: 1.3;
}
