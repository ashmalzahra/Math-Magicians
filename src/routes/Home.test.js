import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Home from './Home';
import '@testing-library/jest-dom';

describe('Checking page deployment', () => {
  test('Home component matches snapshot', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders Home component', () => {
    render(<Home />);

    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(screen.getByText(/Welcome to our page!/)).toBeInTheDocument();
  });
});
