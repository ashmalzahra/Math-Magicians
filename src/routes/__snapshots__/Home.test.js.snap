// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking page deployment Home component matches snapshot 1`] = `
<div>
  <h3
    className="home-text"
  >
    Welcome to our page!
  </h3>
  <p>
    Mathematics is an area of knowledge that includes the topics of numbers, formulas and related structures, shapes and the spaces in which they are contained, and quantities and their changes. These topics are represented in modern mathematics with the major subdisciplines of number theory, algebra, geometry, and analysis, respectively. There is no general consensus among mathematicians about a common definition for their academic discipline.
  </p>
  <p>
    Historically, the concept of a proof and its associated mathematical rigour first appeared in Greek mathematics, most notably in Euclid's Elements. Since its beginning, mathematics was essentially divided into geometry and arithmetic the manipulation of natural numbers and fractions, until the 16th and 17th centuries, when algebra and infinitesimal calculus were introduced as new areas. Since then, the interaction between mathematical innovations and scientific discoveries has led to a rapid lockstep increase in the development of both. At the end of the 19th century, the foundational crisis of mathematics led to the systematization of the axiomatic method, which heralded a dramatic increase in the number of mathematical areas and their fields of application. The contemporary Mathematics Subject Classification lists more than 60 first-level areas of mathematics.
  </p>
</div>
`;
